implants/golemv2/src/main.rs

```diff
@@ -0,0 +1,185 @@
+        )
+        .arg(
+            Arg::new("interactive")
+                .help("Run the interactive REPL")
+                .short('i')
+                .required(false),
+        )
+        .arg(
+            Arg::new("assets")
+                .short('a')
+                .long("assets")
+                .value_name("SOURCE")
+                .action(ArgAction::Append)
+                .help("Asset source (directory, Tavern URL, or ZIP)"),
+        )
+        .arg(
+            Arg::new("embedded")
+                .short('e')
+                .long("embedded")
+                .action(ArgAction::SetTrue)
+                .help("Use embedded assets alongside asset sources"),
+        )
+        .arg(
+            Arg::new("dump")
+                .short('d')
+                .long("dump")
+                .action(ArgAction::SetTrue)
+                .help("Dump tomes to be run and assets"),
+        )
+        .get_matches();
+
+    //let mut parsed_tomes: Vec<ParsedTome> = Vec::new();
+    let mut locker = StdAssetsLibrary::new();
+
+    let asset_directories: Vec<String> = matches
+        .get_many::<String>("assets")
+        .unwrap_or_default()
+        .cloned()
+        .collect();
+
+    // If we have specified asset sources, we need to manually include embedded
+    if asset_directories.len() == 0 || matches.get_flag("embedded") {
+        let backend = EmbeddedAssets::<GolemEmbeddedAssets>::new();
+        locker.add(Arc::new(backend))?;
+    }
+    // Load all the asset directories into the locker
+    for dir in asset_directories {
+        let backend = DirectoryAssetBackend::new(&dir)?;
+        locker.add(Arc::new(backend))?;
+    }
+
+    let mut parsed_tomes: Vec<ParsedTome> = Vec::new();
+    // Input overrides the main.eldritch files in the assets
+    if matches.contains_id("INPUT") {
+        // Read Tomes from the input
+        let tome_files = matches.try_get_many::<String>("INPUT").unwrap().unwrap();
+        for tome in tome_files {
+            let tome_path = tome.to_string().clone();
+            // Read the specified tome off of the disk first, if that fails. try to get it from the asset locker
+            let tome_contents = fs::read_to_string(&tome_path)
+                .map_err(|_| ())
+                .or_else(|_| {
+                    locker
+                        .read(tome_path.clone())
+                        .map_err(|_| anyhow::anyhow!("Error: No such file or asset"))
+                })?;
+
+            parsed_tomes.push(ParsedTome {
+                name: tome_path.clone(),
+                eldritch: tome_contents,
+            });
+        }
+    }
+    // If we havent specified tomes in INPUT, we need to look through the asset locker for tomes to run
+    if parsed_tomes.len() == 0 {
+        match locker.list() {
+            Ok(assets) => {
+                for asset in assets {
+                    if asset.ends_with("main.eldritch") || asset.ends_with("main.eldr") {
```

Comment:
Idk if we want to add .eldr as a new extension. That seems like a separate discussion.